src/commands/publish.rs

```diff
@@ -179,21 +279,192 @@ pub fn publish(publish_opts: PublishOpt) -> anyhow::Result<()> {
     assert!(archive_path.is_file());
 
     if !publish_opts.dry_run {
-        let _response: publish_package_mutation::ResponseData =
-            execute_query_modifier(&q, |f| f.file(archive_name, archive_path).unwrap())
-                .map_err(on_error)?;
+        let _response: publish_package_mutation::ResponseData = execute_query_modifier(&q, |f| {
+            f.file(archive_name.to_string(), archive_path).unwrap()
+        })
+        .map_err(on_error)?;
     }
 
     println!(
         "Successfully published package `{}@{}`",
         package.name, package.version
     );
 
-    if publish_opts.dry_run {
-        info!(
-            "Publish succeeded, but package was not published because it was run in dry-run mode"
+    Ok(())
+}
+
+#[allow(clippy::too_many_arguments)]
+fn try_chunked_uploading(
+    package: &Package,
+    manifest_string: &String,
+    license_file: &Option<String>,
+    readme: &Option<String>,
+    archive_name: &String,
+    archive_path: &PathBuf,
+    maybe_signature_data: &SignArchiveResult,
+    archived_data_size: u64,
+) -> Result<(), anyhow::Error> {
+    let maybe_signature_data = match maybe_signature_data {
+        SignArchiveResult::Ok {
+            public_key_id,
+            signature,
+        } => {
+            info!(
+                "Package successfully signed with public key: \"{}\"!",
+                &public_key_id
+            );
+            Some(publish_package_mutation_chunked::InputSignature {
+                public_key_key_id: public_key_id.to_string(),
+                data: signature.to_string(),
+            })
+        }
+        SignArchiveResult::NoKeyRegistered => {
+            // TODO: uncomment this when we actually want users to start using it
+            //warn!("Publishing package without a verifying signature. Consider registering a key pair with wapm");
+            None
+        }
+    };
+
+    println!("{} {} Uploading...", style("[1/2]").bold().dim(), UPLOAD);
+
+    let get_google_signed_url = GetSignedUrl::build_query(get_signed_url::Variables {
+        name: package.name.to_string(),
+        version: package.version.to_string(),
+    });
+
+    let _response: get_signed_url::ResponseData =
+        execute_query_modifier(&get_google_signed_url, |f| {
+            f.file(archive_name.clone(), archive_path.clone()).unwrap()
+        })?;
+
+    let url = _response.url.ok_or({
+        let e = anyhow!(
+            "could not get signed url for package {}@{}",
+            package.name,
+            package.version
         );
+        #[cfg(feature = "telemetry")]
+        sentry::integrations::anyhow::capture_anyhow(&e);
+        e
+    })?;
```

Comment:
This will ping sentry **every** time someone uploads a package, not just when we don't get a signed URL back.

You want to use `url.ok_or_else(|| { ... })` with a closure that should be called when the url isn't present. Using `url.ok_or(/* some chunk of code */) `will execute the chunk of code all the time because `ok_or()` is a method, and as part of calling that method we always need to evaluate its arguments.

Clippy has [a good explanation](https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call) for what is going on.

```suggestion
    let url = _response.url.ok_or_else(|| {
        let e = anyhow!(
            "could not get signed url for package {}@{}",
            package.name,
            package.version
        );
        #[cfg(feature = "telemetry")]
        sentry::integrations::anyhow::capture_anyhow(&e);
        e
    })?;
```